src/plugins/ethereum/ethAddress.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+
+import {isAddress, toChecksumAddress} from "web3-utils";
+
+import * as C from "../../util/combo";
+import {compatibleParser} from "../../util/compat";
+
+export opaque type EthAddress: string = string;
+
+export function parseAddress(s: string): EthAddress {
+  if (!isAddress(s)) {
+    throw new Error(`not a valid ethereum address: ${s}`);
+  }
+  return toChecksumAddress(s);
+}
+
+// Utilized for a more readable and recognizable address. This is
+// needed to allow the address to fit within the Identity name requirements
+export function truncateEthAddress(address: EthAddress): string {
+  const prefix = address.slice(0, 6);
+  const suffix = address.slice(-4);
+  return `${prefix}...${suffix}`;
+}
+
+export const ethAddressParser: C.Parser<EthAddress> = C.fmap(
+  C.string,
+  parseAddress
+);
+
+export const COMPAT_INFO = {
+  type: "sourcecred/ethAddress",
+  version: "0.0.1",
+};
+
+const addressEntriesParser = C.array(ethAddressParser);
+
+export const parser: C.Parser<
+  Array<EthAddress>
+> = compatibleParser(COMPAT_INFO.type, {"0.0.1": addressEntriesParser});
```

Comment:
What exactly is the purpose of this parser? Why are we storing ETH addresses in an array like this separate from the ledger? Is it supposed to be a whitelist or something?

Curious about the value this data structure provides

Reply:
On behalf of @topocount (correct me if mistaken): this plugin loads data
entirely from its config. Contrast:

  - The GitHub/Discourse/Discord plugins read their config to determine
    a server/repo to hit, and load data by querying it.
  - The Ethereum plugin reads its config to determine addresses, and
    loads data by simply emitting that set of addresses plus node and
    edge structure derived from it.

The PR description implies this in the first two paragraphs, I think. In
general I think this kind of thing is nice to have a brief design doc
for: say, a Discourse post describing the structure and purpose of this
plugin.

Reply:
@hammadj check out the added detail in the PR description